olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h

```diff
@@ -97,6 +104,11 @@ class CORE_API TaskScheduler {
    * kept. Once this method is called, you own the task.
    */
   virtual void EnqueueTask(CallFuncType&&) = 0;
+
+  virtual void EnqueueTask(CallFuncType&& func, Priority priority) {
```

Comment:
I would not use enum here, as this again limits us to only 3 values. Use uint32_t or uint16_t and provide the enum just as a helper.